### PR TITLE
Make virtualization tests by aginies usable for openSUSE

### PR DESCRIPTION
**Pull request by okurz**
The test modules within "tests/virtualization" were never executed on
production. QA SLE has used a different approach in parallel to run
virtualization tests for SLE focussing on bare-metal testing over an IPMI+ssh
connection.

The first modules of the test modules in "tests/virtualization" have been
reworked to run based on current openSUSE Tumbleweed currently passing tests.

Verification run: http://lord.arch/tests/59

Needles PR:
https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/296

Related progress issue: https://progress.opensuse.org/issues/18956

### CODE REVIEW

**Comment by okurz** [comment]
It seems I can't continue because of exactly the issue that was also reported caused by the move to python3. That was expected, right? ;)

exc_referenced_before_assignment

So the test should be very helpul up to the current state and then we can extend them further.

**Comment by aginies** [comment]
yes its due to py3 move.

**Comment by okurz** [comment]
someone can review and wants to merge?

File: lib/virtmanager.pm [hunk 1]
```diff
@@ -4,216 +4,175 @@ use strict;
...
     valid_step();
     # always close this windows to restart from scratch
-    send_key "ctrl-w";
+    send_key 'ctrl-w';
 }
 
 sub loop_down {
     my ($count) = @_;
-    for (1 .. $count) { send_key "down"; }
+    send_key 'down' for (1 .. $count);
```
Comment (asmorodskyi) [inline lib/virtmanager.pm]: not really against , but just curious what the reason for this change ?

**Comment by asmorodskyi** [comment]
looks awesome to me ;)

File: lib/virtmanager.pm (hunk 1 above)
Comment (okurz) [reply lib/virtmanager.pm]: less characters to type ;)

**Comment by asmorodskyi** [comment]
only thing that concerns me - proof run is actually failing . Is it expected ? Currently it's failing on asserting screen with "virt-manager_details" tag I fail to find needle with such tag at all

File: lib/virtmanager.pm (hunk 1 above)
Comment (asmorodskyi) [reply lib/virtmanager.pm]: hard to argue :)

**Comment by coolo** [comment]
https://openqa.suse.de/tests/1315739#step/yast_virtualization/11 - please fix/needle

**Pull request merged**